logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+/** @file db_api.c
+ *  @brief This is the file containing the database API
+ *
+ *  @author Dimitris Pantazis
+ */
+
+#include "../daemon/common.h"
+#include "db_api.h"
+#include <inttypes.h>
+#include <stdio.h>
+#include "circular_buffer.h"
+#include "compression.h"
+#include "helper.h"
+#include "lz4.h"
+#include "parser.h"
+
+#define MAIN_DB "main.db" /**< Primary DB with just 1 table - MAIN_COLLECTIONS_TABLE **/
+#define MAIN_COLLECTIONS_TABLE "LogCollections"
+#define BLOB_STORE_FILENAME "logs.bin"
+#define METADATA_DB_FILENAME "metadata.db"
+#define LOGS_TABLE "Logs"
+#define BLOBS_TABLE "Blobs"
+
+static uv_loop_t *db_loop;
+static sqlite3 *main_db;
+static char *main_db_dir;  /**< Directory where all the log management databases and log blobs are stored in **/
+static char *main_db_path; /**< Path of MAIN_DB **/
+
+void db_set_lock(uv_mutex_t *db_mut) { 
+	uv_mutex_lock(db_mut); 
+}
+
+void db_release_lock(uv_mutex_t *db_mut) { 
+	uv_mutex_unlock(db_mut);
+}
+
+/**
+ * @brief Throws fatal SQLite3 error
+ * @details In case of a fatal SQLite3 error, the SQLite3 error code will be 
+ * translated to a readable error message and logged to stderr. 
+ * @param[in] rc SQLite3 error code
+ * @param[in] line_no Line number where the error occurred 
+ */
+static inline void fatal_sqlite3_err(int rc, int line_no){
+	debug(D_LOGS_MANAG, "SQL ERROR:%s", sqlite3_errstr(rc));
```

Comment:
We almost have duplicated information inside this function, probably we can remove the `debug` message. :thinking:

Reply:
Done.